jbmc/src/java_bytecode/README.md

```diff
@@ -39,9 +39,21 @@ To be documented.
 
 \subsection java-bytecode-pointer-type-selection Pointer type selection
 
-To be documented.
-
-\subsection java-bytecode-genereic-substitution Generic substitution
+In Java, all variables that are of a non-primitive type are pointers to
+an object. When initializing such variables, e.g., as input arguments for the
+method under test, we may need to select the correct pointer type. For example,
+abstract classes can be replaced with (any of) their concrete implementations
+and type parameters in generic types can be replaced with their specialized
+types.
+
+The class \ref select_pointer_typet offers the basic interface for this
+functionality, in particular see
+\ref select_pointer_typet::convert_pointer_type. Note that this class only
+implements generic substitution (see \ref java-bytecode-generic-substitution),
+derived classes can override this behaviour to provide more
```

Comment:
:pick: As in the other commit, should probably use American spelling here.

Reply:
Done